src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -533,8 +533,8 @@ private IObjectOrCollectionInitializerExpression CreateBoundCollectionInitialize
         private IMemberReferenceExpression CreateBoundObjectInitializerMemberOperation(BoundObjectInitializerMember boundObjectInitializerMember)
         {
             Lazy<IOperation> instance = new Lazy<IOperation>(() => new InstanceReferenceExpression(
-                InstanceReferenceKind.Implicit,
-                _semanticModel,
+                isBaseReference: false,
```

Comment:
>isBaseReference: false [](start = 16, length = 22)

We might want to mention this use of IInstanceReferenceExpression in its doc comment. Right now it says: "Represents a C# this or base expression, or a VB Me, MyClass, or MyBase expression." Which doesn't cover this use case.

Reply:
I added a test for this, and indeed the instance reference here will have the base type. @mavasani @AlekseyTs, what is the behavior we want here? It seems to me like this behavior is fine, as this is initializing a member that is in the base class, not the derived class. However, I can see the argument for the other way around.

Reply:
I guess we have to clarify the meaning of IInstanceReferenceExpression in for all possible use contexts. Let's discuss this at design meeting.

Reply:
@AlekseyTs filed https://github.com/dotnet/roslyn/issues/21769. I'd like to merge this PR and address anything we decide at the design meeting in a follow-up PR.